drivers/include/tca6507.h

```diff
@@ -0,0 +1,257 @@
+  * @ingroup     drivers_saul
+  * @defgroup    drivers_tca6507 TCA6507 I2C Led controller
+  * @brief       Public API for TCA6507 I2C Led controller
+  * @author      Noel Le Moult <noel.lemoult@dfxlab.fr>
+  * @author      William MARTIN <william.martin@power-lan.com>
+  * @file
+  * @{
+  */
+
+#ifndef TCA6507_H
+#define TCA6507_H
+
+#include "periph/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Driver return codes
+ */
+enum {
+    TCA6507_OK,                 /**< All OK */
+    TCA6507_ERR_NODEV,          /**< No valid device found on I2C bus */
+    TCA6507_ERR_I2C             /**< An error occured when reading/writing on I2C bus */
+};
+
+
+/**
+ * @brief Bank (pwm) define
+ */
+enum {
+    TCA6507_BANK0,
+    TCA6507_BANK1,
+    TCA6507_BANK_COUNT
+};
+
+/**
+ * @brief Fadding times
+ */
+enum {
+    TCA6507_TIME_0_MS,
+    TCA6507_TIME_64_MS,
+    TCA6507_TIME_128_MS,
+    TCA6507_TIME_192_MS,
+    TCA6507_TIME_256_MS,
+    TCA6507_TIME_384_MS,
+    TCA6507_TIME_512_MS,
+    TCA6507_TIME_768_MS,
+    TCA6507_TIME_1024_MS,
+    TCA6507_TIME_2536_MS,
+    TCA6507_TIME_2048_MS,
+    TCA6507_TIME_3072_MS,
+    TCA6507_TIME_4096_MS,
+    TCA6507_TIME_5760_MS,
+    TCA6507_TIME_8128_MS,
+    TCA6507_TIME_16320_MS,
+    TCA6507_TIME_COUNT
+};
+
+/**
+ * @brief Brightness
+ */
+enum {
+    TCA6507_BRIGHTNESS_0_PCENT,
+    TCA6507_BRIGHTNESS_6_PCENT,
+    TCA6507_BRIGHTNESS_13_PCENT,
+    TCA6507_BRIGHTNESS_20_PCENT,
+    TCA6507_BRIGHTNESS_26_PCENT,
+    TCA6507_BRIGHTNESS_33_PCENT,
+    TCA6507_BRIGHTNESS_40_PCENT,
+    TCA6507_BRIGHTNESS_46_PCENT,
+    TCA6507_BRIGHTNESS_53_PCENT,
+    TCA6507_BRIGHTNESS_60_PCENT,
+    TCA6507_BRIGHTNESS_66_PCENT,
+    TCA6507_BRIGHTNESS_73_PCENT,
+    TCA6507_BRIGHTNESS_80_PCENT,
+    TCA6507_BRIGHTNESS_86_PCENT,
+    TCA6507_BRIGHTNESS_93_PCENT,
+    TCA6507_BRIGHTNESS_100_PCENT,
+    TCA6507_BRIGHTNESS_COUNT
+
+};
+
+/**
+ * @brief Device initialization parameters.
+ */
+typedef struct {
+    i2c_t i2c_dev;              /**< I2C bus the sensor is connected to */
+    uint8_t address;            /**< sensor address */
+} tca6507_params_t;
+
+/**
+ * @brief TCA6507 device descriptor.
+ */
+typedef struct {
+    tca6507_params_t params;     /**< Device parameters */
+} tca6507_t;
+
+/**
+ * @brief   Initialize and reset the sensor.
+ *
+ * @param[in] dev           device descriptor
+ * @param[in] params        initialization parameters
+ *
+ * @return                  TCA6507_OK on successful initialization
+ * @return                  TCA6507_ERR_NODEV on device test error
+ * @return                  TCA6507_ERR_I2C on I2C bus error
+ */
+int tca6507_init(tca6507_t *dev, const tca6507_params_t *params);
+
+/**
+ * @brief   Turn on a led
+ *
+ * @param[in] dev           device descriptor
+ * @param[in] led           The led index
+ * @param[in] bank          The bank to use (intensity)
+ *
+ * @return                  TCA6507_OK on successful initialization
```

Comment:
All functions in the API returns `TCA6507_OK` *on successful initialization*. Copy-paste ? :D